content/0000-puppet-integration/rfc.md

```diff
@@ -0,0 +1,420 @@
+    - node-a.example.com
+    - node-b.example.com
+  noOp: false
+  debug: false
+  trace: false
+  evalTrace: false
+  ```
+
+  Outputs:
+
+  ```yaml
+  jobID: https://orchestrator.example.com:8143/orchestrator/v1/jobs/1234
+  outcome: finished
+  ```
+
+## Drawbacks
+
+This design is intentionally _very_ Puppet-specific and even then does not
+provide integration with much of any PE or Puppet functionality beyond run
+execution. So this comes with the cost of excluding early users that might seek
+a more generic on-premise solution.
+
+Even in its relative simplicity, it requires a decent amount of work to add and
+maintain new API endpoints and connection types in the Relay service. I worry
+somewhat about making radical changes to the connection types in particular in
+the future, and we may need to consider implementing versioning for connections
+sooner than anticipated.
+
+## Rationale and alternatives
+
+### Why is this design the best in the space of possible designs?
+
+Truthfully, this likely isn't the best technical design for this need. But it is
+cheap to design, implement, test, validate, and support. In this case, since our
+primary objective is to validate the product integration and use case, we accept
+the trade-off in reusability and extensibility.
+
+### What other designs have been considered and what is the rationale for not choosing them?
+
+An obvious design choice is the full implementation of the on-premise Relay Core
+offering. However, we don't yet have a clear way to bundle it with PE (much less
+open source Puppet) as it requires an operating Kubernetes cluster to work. In
+the future, we do see this being an appealing choice for PE customers,
+especially those with strict data locality requirements.
+
+We considered a long-running Web server or client process that we proxy to our
+service to provide direct RPCs. This could be, for example, something like
+[Inlets](https://github.com/inlets/inlets) or a WebSocket-based protocol.
+However, we decided the maintenance requirements and design complexity were too
+high for this proof of concept.
+
+### What is the impact of not doing this?
+
+We may not get the initial information we need to validate our Puppet
+integration use case. We might also make poor decisions in the future when
+deciding how to implement our generic on-premise solution.
+
+### What specific risks are associated with this design?
+
+As stated, this is intended to be alpha-quality software. We have a slight risk
+of not adequately communicating this and being forced to maintain the APIs
+defined here beyond their usable lifetimes.
+
+While this design does not require firewall changes for most on-premise networks
+that allow outbound internet traffic, it cannot work in a fully air-gapped
+environment. It also exposes internal PE node data to our SaaS service if the
+report processor plugin is used. Some customers may not be able to use this
+module for regulatory or compliance reasons.
```

Comment:
Especially considering sensitive information in the logs as in the gotcha section of https://puppet.com/blog/encrypt-your-data-using-hiera-eyaml/